racket/src/racket/src/jitinline.c

```diff
@@ -2532,9 +2569,47 @@ static int generate_binary_char(mz_jit_state *jitter, Scheme_App3_Rec *app,
     /* Extract character value */
     jit_ldxi_i(JIT_R0, JIT_R0, (intptr_t)&SCHEME_CHAR_VAL((Scheme_Object *)0x0));
     jit_ldxi_i(JIT_R1, JIT_R1, (intptr_t)&SCHEME_CHAR_VAL((Scheme_Object *)0x0));
-    ref = jit_bner_i(jit_forward(), JIT_R0, JIT_R1);
+
+    /* FIXME: why are these tests all inverted? */
```

Comment:
Because "(if <something> <t> <f>)" turns into "if (not <something>) goto false; <t>; jump to end; false: <f>; end:"